### PR TITLE
[RHELC-723] Remove RegistrationCommand.args property

### PR DESCRIPTION
**Pull request by pr-watson**
Removed RegistrationCommand.args property and modified the log messages to take the arguments from self rather than the RegistrationCommand.args property https://issues.redhat.com/browse/RHELC-723

Signed-off-by: Preston Watson <prwatson@redhat.com>

Checklist
- [x] PR meets acceptance criteria specified in the Jira issue
- [x] PR has been tested manually in a VM (either author or reviewer)
- [x] Jira issue has been made public if possible
- [x] `[RHELC-]` is part of the PR title 
- [x] Code and tests are documented properly
- [x] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending`

### CODE REVIEW

**Comment by pr-watson** [comment]
Here is a snippet of the logging messages after the change:

[12/08/2022 21:15:55] DEBUG - Connecting to the private DBus

Registering via username/password:

Organization: None

Username: c2r_main

Password: *****

Register Options: dbus.Dictionary({'force': True}, signature=dbus.Signature('sv'), variant_level=1)

Connection Options: dbus.Dictionary({'host': 'subscription.rhsm.stage.redhat.com'}, signature=dbus.Signature('sv'), variant_level=1)

Locale Setting: C


Setting RHSM connection configuration.

File: convert2rhel/subscription.py
Comment (r0x0d) [inline convert2rhel/subscription.py:479]: @pr-watson, I would modify this logging a bit and make it a bit simpler to read and to maintain later, maybe something like this: 

 ```suggestion
                    loggerinst.info("Organization: %s", self.org)
                    loggerinst.info("Activation Key: %s", "*" * 5)
                    loggerinst.info("Connection Options: %s", self.connection_opts)
                    loggerinst.info("Locale settings: %s", i18n.SUBSCRIPTION_MANAGER_LOCALE)
```

I would say it's totally fine to have multiple log calls as far as they are easier to ready and maintain... What do you think? 

NOTE: This also applies to the logging text above that you have in the same way.

**Review by r0x0d — COMMENTED** [review]
Thank you for your first PR, @pr-watson! It's looking pretty good, only a couple of small things that could be modified! 

If you can, please, fix the `CodeQL` suggestions in your PR (mainly, removing the unused import of `urllib` as you removed the whole piece of code that made use of them) just so we don't have any more warnings/suggestions to our bucket as the PRs get merged. 

On another note, I read in the description of that issue that this is supposed to use the `utils.hide_secrets()` function to make the output obfuscated for password/activation_key, but unfortunately that function is supposed to work with CLI parameters rather than any other type of text. @abadger do you think it's worth it to enhance that a bit to also accept other type of data, this way we don't need to explicitly call `"*" * 5` here and there whenever we want to work with an obfuscated output. 

In any case, looking good and waiting for your reply, @pr-watson!

**Comment by pr-watson** [comment]
@r0x0d I'll also get rid of the unused import for urllib

File: convert2rhel/subscription.py
Comment (r0x0d) [inline convert2rhel/subscription.py:435]: I think we want to change it here to use `username` and `password` instead of `activation key`, as it seems that the `activation key` one is only the one that happens in the `else` statement... 

```suggestion
                    loggerinst.info("Username: %s", self.username)
                    loggerinst.info("Password: %s", "*" * 5)
```

File: convert2rhel/subscription.py
Comment (r0x0d) [inline convert2rhel/subscription.py:434]: I'm a bit unsure if we have to print `Organization:` here... Maybe we do because we set that in the line 439 :thinking:  I guess it would be nice to confirm with @abadger since he did the code a while ago.

Comment (r0x0d) [reply convert2rhel/subscription.py:434]: Also, if we do, I would do conditional to print that only if there is an org present otherwise, I would omit this information as it is not relevant if it's not present, like so: 

```suggestion
                    if self.org:
                        loggerinst.info("Organization: %s", self.org)

```

Comment (r0x0d) [reply convert2rhel/subscription.py:434]: The indentation may be wrong, though...

File: convert2rhel/subscription.py
Comment (pr-watson) [reply convert2rhel/subscription.py:435]: Good catch, i'll change those lines

File: convert2rhel/subscription.py
Comment (pr-watson) [reply convert2rhel/subscription.py:434]: I'll wait on @abadger to respond for this change

Comment (abadger) [reply convert2rhel/subscription.py]: It turns out that organization is needed even for username and password if the username is associated with more than one organization.  We have a ticket open to handle that case: https://issues.redhat.com/browse/RHELC-726

So logging the organization is a good idea here.

**Comment by abadger** [comment]
> On another note, I read in the description of that issue that this is supposed to use the `utils.hide_secrets()` function to make the output obfuscated for password/activation_key, but unfortunately that function is supposed to work with CLI parameters rather than any other type of text. @abadger do you think it's worth it to enhance that a bit to also accept other type of data, this way we don't need to explicitly call `"*" * 5` here and there whenever we want to work with an obfuscated output.

I've been contemplating that over the weekend.  I think it doesn't make sense to enhance that particular function.  The reason is that we don't have a good way to tell that the password is used just from a bare variable.  The function is able to do its job on the command line args because they're key value pairs, where the key tells us that we're dealing with a password or other secret.

We could extract the obfuscation string itself into its own function or a global variable... but I lean towards it not being worth it:
``` python
# As a function:
def obfuscate_value(value):
    return "*" * 5

password = "s3cret"
print("Password %s" % obfuscate_value(password))

# As a global:
SECRET_REPLACEMENT_STRING = "*" * 5

print("Password %s" % SECRET_REPLACEMENT_STRING
```

**Comment by r0x0d** [comment]
> > On another note, I read in the description of that issue that this is supposed to use the `utils.hide_secrets()` function to make the output obfuscated for password/activation_key, but unfortunately that function is supposed to work with CLI parameters rather than any other type of text. @abadger do you think it's worth it to enhance that a bit to also accept other type of data, this way we don't need to explicitly call `"*" * 5` here and there whenever we want to work with an obfuscated output.
> 
> I've been contemplating that over the weekend. I think it doesn't make sense to enhance that particular function. The reason is that we don't have a good way to tell that the password is used just from a bare variable. The function is able to do its job on the command line args because they're key value pairs, where the key tells us that we're dealing with a password or other secret.
> 
> We could extract the obfuscation string itself into its own function or a global variable... but I lean towards it not being worth it:
> 
> ```python
> # As a function:
> def obfuscate_value(value):
>     return "*" * 5
> 
> password = "s3cret"
> print("Password %s" % obfuscate_value(password))
> 
> # As a global:
> SECRET_REPLACEMENT_STRING = "*" * 5
> 
> print("Password %s" % SECRET_REPLACEMENT_STRING
> ```

Makes sense to me.... I was trying to figure out another way of doing it, but it turns out that there are not many cases to use such function, and I think most of the time we will use the `hide_secrets()` the way it is.

**Comment by pr-watson** [comment]
This new patch fixes the broken unit tests, combines two unit tests (test_calling_registration_command_activation_key and test_calling_registration_command_password) into one general unit test. This unit test also checks that Organization is properly logged using caplog.
Thanks @abadger for guiding me through the pytest process

**Comment by SpyTec** [comment]
Did a rebase to be up to date with pipeline and also trigger new integration tests

**Comment by r0x0d** [comment]
@danmyway it seems that we still have the problem in our integration tests, judging by the results here :(

File: convert2rhel/unit_tests/subscription_test.py
Comment (SpyTec) [inline convert2rhel/unit_tests/subscription_test.py:803]: If we don't need it we should remove it
```suggestion
```

**Review by SpyTec — CHANGES_REQUESTED** [review]
One remark otherwise LGTM. Will move to dev complete

Edit: Ah we decided integration tests are not needed. Just have to make sure the existing ones pass

File: convert2rhel/unit_tests/subscription_test.py
Comment (pr-watson) [reply convert2rhel/unit_tests/subscription_test.py]: Just made the change, thanks @SpyTec

**Review by SpyTec — APPROVED** [review]

**Review by r0x0d — APPROVED** [review]
Looks good to me! Thanks for the work here, @pr-watson

File: convert2rhel/subscription.py
Comment (bocekm) [inline convert2rhel/subscription.py:436]: Please replace these two with asterisks based on https://github.com/oamg/convert2rhel/pull/725.

You can perhaps define this `obfuscation_string` as a constant on a module level and then use it in this `subscription.py` module as well as in `utils.py`.
https://github.com/oamg/convert2rhel/blob/v1.1/convert2rhel/utils.py#L722

**Review by bocekm — CHANGES_REQUESTED** [review]

File: convert2rhel/subscription.py
Comment (bocekm) [reply convert2rhel/subscription.py:436]: Or if this is ready to merge, we can do so and I can update #725.

Comment (pr-watson) [reply convert2rhel/subscription.py:436]: @bocekm it is ready to merge, it would be great if you could update #725 thanks!

**Comment by r0x0d** [comment]
The failures here looks fine in my opinion. They don't seem to be related to the changes in the PR. I will leave it up to you guys to decide if you want to merge it right away or wait for an investigation regarding the errors.

File: convert2rhel/subscription.py
Comment (bocekm) [reply convert2rhel/subscription.py:436]: Done.

**Pull request merged**
